### DISCUSSION TITLE
SegResNet_fcon_Dataset

### DISCUSSION THREAD

**Discussion by vivekmodgill** [Project-MONAI/MONAI#4188] [Q&A]
Hi,
We are working with fcon dataset to train a model to detect brain lesions and to do that we are following BratsSegmentation Monai tutorial.
We are stuck with an error for pretty long.
Error: The size of tensor a (60) must match the size of tensor b (59) at non-singleton dimension 3.

Any help would be highly appreciated, feel free to drop your email if you want to try the code with a subset of data.

Thanks.
[MONAI_VS.txt](https://github.com/Project-MONAI/MONAI/files/8574232/MONAI_VS.txt)

**Comment by Nic-Ma** [comment]
Hi @yiheng-wang-nv ,

Could you please help take a look at this question?

Thanks in advance.

**Comment by yiheng-wang-nv** [comment]
Hi @ChaoticNeuron , sorry for late reply. The place that has the issue is: `roi_size=(197, 233, 189)` in `sliding_window_inference`.
In addition, what is the spatial size of your input? I did not find any `resize` or `pad` or `crop` in your transforms.
Usually, we may need to use even numbers as the input size of a network. The reason is that the convolution layers usually use `stride=2`, odd numbers may lead to input and output shape mismatch. You can check the link for more details about how to calculate the input/output size for a conv3d layer:
https://pytorch.org/docs/stable/generated/torch.nn.Conv3d.html

If a network has multiple down sample layers, like 4 in `SegResNet` (it is determined by `blocks_down`) in your case, the input size may need to be divisible by `2**4` which is 16. Therefore, I would suggest to use `roi_size=(192, 224, 192)` for `sliding_window_inference`, and you may also need to do crop/resize/pad during the transforms to control your input size of the network.

**Reply by vivekmodgill** [reply]
Okay, Thank you so much.

**Reply by alcarpo** [reply]
Hi @yiheng-wang-nv, 
We have actually realised the error according to odd numbers in the z (slices) dimensions, thank you so much. 
However, I would like to ask you some troubles we are facing hoping you could bring in a little bright light :)

As @ChaoticNeuron commented, we are dealing with 655 images from the ATLAS dataset with shape (197, 233, 189). We have resized them with the "ResizeWithPadOrCopd" transform to (240, 240, 160) initially and train the model with 150 epochs and 8 filters due to GPU out of memory, and a roi_size suggested of (192, 224, 192) for the sliding_window_inference, but got a mean dice of 0.56. 
Then, we tried to reduce and resize the images to (208, 208, 112) to avoid the CUDA out of memory error commented before, and train the model with 150 epochs, 16 filters, with a roi_size of (224, 224, 128) for the sliding_window_inference, to improve the mean dice, but still got a mean dice of 0.56.


We were wondering if anybody would help us to figure out what we have missed to increase the model performance on the training dataset. We have a GeForce RTX 2080 Ti 11 GB.

I attach the .pynb file we are working with for better comprehension and follow up. 
[05_13_SPECS_MONAI-ATLAS dataset.zip](https://github.com/Project-MONAI/MONAI/files/8685728/05_13_SPECS_MONAI-ATLAS.dataset.zip)
